service/collector/app_resource.go

```diff
@@ -79,24 +113,38 @@ func (c *AppResource) collectAppStatus(ctx context.Context, ch chan<- prometheus
 			app.Status.Version,
 		)
 
-	}
+		if !key.IsCordoned(app) {
+			continue
+		}
 
-}
+		t, err := convertToTime(key.CordonUntil(app))
+		if err != nil {
+			c.logger.Log("level", "warning", "message", "could not convert cordon-until", "stack", fmt.Sprintf("%#v", err))
+			continue
+		}
 
-// Collect is the main metrics collection function.
-func (c *AppResource) Collect(ch chan<- prometheus.Metric) error {
-	ctx := context.Background()
+		ch <- prometheus.MustNewConstMetric(
+			appCordonExpireTimeDesc,
+			prometheus.GaugeValue,
+			float64(t.Unix()),
+			key.AppName(app),
+		)
+	}
+	return nil
+}
 
-	c.logger.LogCtx(ctx, "level", "debug", "message", "collecting metrics")
+func convertToTime(datetime string) (time.Time, error) {
+	layout := "2006-01-02T15:04:05"
```

Comment:
In the unit test it has milliseconds `2020-01-02T15:04:05.000`. Is that required? 

We are going to set the annotation in opsctl but I'd like to be clearer on the datetime format.

Reply:
As we discussed below, we would strip milliseconds part.